src/Microsoft.Diagnostics.Runtime/src/MacOS/MachOModule.cs

```diff
@@ -94,18 +94,36 @@ public bool TryLookupSymbol(string symbol, out ulong address)
             if (symbol is null)
                 throw new ArgumentNullException(nameof(symbol));
 
+            if (_stringTableAddress != 0)
+            {
+                // First, search just the "external" export symbols 
+                if (TryLookupSymbol(_dysymtab.iextdefsym, _dysymtab.nextdefsym, symbol, out address))
+                {
+                    return true;
+                }
+
+                // If not found in external symbols, search all of them
+                if (TryLookupSymbol(0, _symtab.NSyms, symbol, out address))
+                {
+                    return true;
+                }
+            }
+
             address = 0;
+            return false;
+        }
 
-            if (_stringTableAddress == 0)
-                return false;
+        private bool TryLookupSymbol(uint start, uint nsyms, string symbol, out ulong address)
+        {
+            address = 0;
 
             NList64[]? symTable = ReadSymbolTable();
             if (symTable is null)
                 return false;
 
-            for (int i = 0; i < _dysymtab.nextdefsym; i++)
+            for (uint i = 0; i < nsyms; i++)
```

Comment:
This doesn't fully bounds check `symTable`.  Perhaps something like this, for defensive coding purposes?

```suggestion
            for (uint i = 0; i < nsyms && start + i < symTable.Length; i++)
```

Reply:
Good catch.  I pushed the changes.